### PR TITLE
fix(email notification): html entity decode

### PR DESCRIPTION
**Pull request by Rom1-B**
The `&` in the notification settings was displayed as `&#38;` in received mails.

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !30570

### CODE REVIEW

**Review by cconard96 — APPROVED** [review]

File: src/MailCollector.php [hunk 1]
```diff
@@ -2072,12 +2072,12 @@ public function sendMailRefusedResponse($to = '', $subject = '')
 
         $mmail = new GLPIMailer();
         $mmail->AddCustomHeader("Auto-Submitted: auto-replied");
-        $mmail->SetFrom($CFG_GLPI["admin_email"], $CFG_GLPI["admin_email_name"]);
+        $mmail->SetFrom($CFG_GLPI["admin_email"], html_entity_decode($CFG_GLPI["admin_email_name"]));
```
Comment (cedric-anne) [inline src/MailCollector.php:2075]:
```suggestion
        $mmail->SetFrom($CFG_GLPI["admin_email"], Sanitizer::decodeHtmlSpecialChars($CFG_GLPI["admin_email_name"]));
```

File: src/MailCollector.php [hunk 2]
```diff
@@ -2072,12 +2072,12 @@ public function sendMailRefusedResponse($to = '', $subject = '')
...
         $mmail->AddAddress($to);
        // Normalized header, no translation
         $mmail->Subject  = 'Re: ' . $subject;
         $mmail->Body     = __("Your email could not be processed.\nIf the problem persists, contact the administrator") .
-                         "\n-- \n" . $CFG_GLPI["mailing_signature"];
+                         "\n-- \n" . html_entity_decode($CFG_GLPI["mailing_signature"]);
```
Comment (cedric-anne) [inline src/MailCollector.php:2080]:
```suggestion
                         "\n-- \n" . Sanitizer::decodeHtmlSpecialChars($CFG_GLPI["mailing_signature"]);
```

File: src/NotificationEventMailing.php [hunk 3]
```diff
@@ -172,10 +172,10 @@ public static function send(array $data)
                     }
                 }
 
-                $mmail->SetFrom($current->fields['sender'], $current->fields['sendername']);
+                $mmail->SetFrom($current->fields['sender'], html_entity_decode($current->fields['sendername']));
```
Comment (cedric-anne) [inline src/NotificationEventMailing.php:175]:
```suggestion
                $mmail->SetFrom($current->fields['sender'], Sanitizer::decodeHtmlSpecialChars($current->fields['sendername']));
```

File: src/NotificationEventMailing.php [hunk 4]
```diff
@@ -172,10 +172,10 @@ public static function send(array $data)
...
 
                 if ($current->fields['replyto']) {
-                    $mmail->AddReplyTo($current->fields['replyto'], $current->fields['replytoname']);
+                    $mmail->AddReplyTo($current->fields['replyto'], html_entity_decode($current->fields['replytoname']));
```
Comment (cedric-anne) [inline src/NotificationEventMailing.php:178]:
```suggestion
                    $mmail->AddReplyTo($current->fields['replyto'], Sanitizer::decodeHtmlSpecialChars($current->fields['replytoname']));
```

File: src/NotificationEventMailing.php [hunk 5]
```diff
@@ -385,7 +385,7 @@ public static function send(array $data)
                     $mmail->AltBody .= $text;
                 }
 
-                $mmail->AddAddress($recipient, $current->fields['recipientname']);
+                $mmail->AddAddress($recipient, html_entity_decode($current->fields['recipientname']));
```
Comment (cedric-anne) [inline src/NotificationEventMailing.php:388]:
```suggestion
                $mmail->AddAddress($recipient, Sanitizer::decodeHtmlSpecialChars($current->fields['recipientname']));
```

File: src/NotificationMailing.php [hunk 6]
```diff
@@ -103,9 +103,9 @@ public static function testNotification()
         $mmail->AddCustomHeader("Auto-Submitted: auto-generated");
        // For exchange
         $mmail->AddCustomHeader("X-Auto-Response-Suppress: OOF, DR, NDR, RN, NRN");
-        $mmail->SetFrom($sender['email'], $sender['name'] ?? '', false);
+        $mmail->SetFrom($sender['email'], html_entity_decode($sender['name'] ?? ''), false);
```
Comment (cedric-anne) [inline src/NotificationMailing.php:106]:
```suggestion
        $mmail->SetFrom($sender['email'], Sanitizer::decodeHtmlSpecialChars($sender['name'] ?? ''), false);
```

File: src/NotificationMailing.php [hunk 7]
```diff
@@ -103,9 +103,9 @@ public static function testNotification()
...
 
-        $text = __('This is a test email.') . "\n-- \n" . $CFG_GLPI["mailing_signature"];
+        $text = __('This is a test email.') . "\n-- \n" . html_entity_decode($CFG_GLPI["mailing_signature"]);
```
Comment (cedric-anne) [inline src/NotificationMailing.php:108]:
```suggestion
        $text = __('This is a test email.') . "\n-- \n" . Sanitizer::decodeHtmlSpecialChars($CFG_GLPI["mailing_signature"]);
```

File: src/NotificationMailing.php [hunk 8]
```diff
@@ -114,7 +114,7 @@ public static function testNotification()
             $text .= "\n" . sprintf(__('Original email address was %1$s'), $CFG_GLPI['admin_email']);
         }
 
-        $mmail->AddAddress($recipient, $CFG_GLPI["admin_email_name"]);
+        $mmail->AddAddress($recipient, html_entity_decode($CFG_GLPI["admin_email_name"]));
```
Comment (cedric-anne) [inline src/NotificationMailing.php:117]:
```suggestion
        $mmail->AddAddress($recipient, Sanitizer::decodeHtmlSpecialChars($CFG_GLPI["admin_email_name"]));
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]
When a previously sanitized content must be "unsanitized", it is preferable to use the corresponding `Sanitizer` methods.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**